src/posix/platform/netif.cpp

```diff
@@ -384,12 +420,23 @@ static void UpdateUnicastLinux(const otIp6AddressInfo &aAddressInfo, bool aIsAdd
         AddRtAttr(&req.nh, sizeof(req), IFA_CACHEINFO, &cacheinfo, sizeof(cacheinfo));
     }
 
-#if OPENTHREAD_POSIX_CONFIG_NETIF_PREFIX_ROUTE_METRIC > 0
-    if (aAddressInfo.mScope > ot::Ip6::Address::kLinkLocalScope)
+#if OPENTHREAD_POSIX_CONFIG_INSTALL_OMR_ROUTES_ENABLE
+    if (IsOmrAddress(aInstance, aAddressInfo))
     {
-        AddRtAttrUint32(&req.nh, sizeof(req), IFA_RT_PRIORITY, OPENTHREAD_POSIX_CONFIG_NETIF_PREFIX_ROUTE_METRIC);
+        // Remove prefix route for OMR address if `OPENTHREAD_POSIX_CONFIG_INSTALL_OMR_ROUTES_ENABLE` is enabled to
+        // avoid having two routes.
+        AddRtAttrUint32(&req.nh, sizeof(req), IFA_FLAGS, IFA_F_NOPREFIXROUTE);
     }
+    else
+#endif
+    {
+#if OPENTHREAD_POSIX_CONFIG_NETIF_PREFIX_ROUTE_METRIC > 0
+        if (aAddressInfo.mScope > ot::Ip6::Address::kLinkLocalScope)
+        {
+            AddRtAttrUint32(&req.nh, sizeof(req), IFA_RT_PRIORITY, OPENTHREAD_POSIX_CONFIG_NETIF_PREFIX_ROUTE_METRIC);
+        }
 #endif
+    }
```

Comment:
Good point. The current code is working.
I think it's still fine (even better) to specify the metric when removing the OMR route? It make sure we are removing the correct route even when there are multiple OMR routes (some may not be added by OT).

For example:
```
$ sudo ip -6 route | grep 2005
2005::/64 dev eno1 metric 1 pref medium
2005::/64 dev eno1 metric 2 pref medium
$ sudo ip -6 route del 2005::/64 metric 2
$ sudo ip -6 route | grep 2005
2005::/64 dev eno1 metric 1 pref medium
```

It would remove the `metric 1` route if metric is not specified in `ip -6 route del`.

Reply:
Update: Yes you are right. No need to specify `noprefixroute` when removing. But it did work.